### DISCUSSION TITLE
⏳ 🔗 Elements Manager: List Pages and Templates Where Found Elements are Used

### DISCUSSION THREAD

**Discussion by micmiller-apptio** [elementor/elementor#24748] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

The Elements Manager is a welcome addition to Elementor, however it is missing the ability to list and drill into the places where widgets are used. When doing things like consolidating widget usage or converting content from one set of widgets to another, it is essential to know where (on what pages a templates) a widget is being used.

### Describe the solution you'd like.

I could click a link in a row for a widget to get a list of places where that widget is used, similar to clicking on a taxonomy term and seeing all the posts that use that term. Beyond that, it would be helpful to have an "Edit with Elementor" link for each of the listed pages/posts/templates that would be displayed.

### Describe alternatives you've considered.

I have used a third-party plugin, Better Usage Statistics for Elementor, which includes a display of all pages and templates and the widgets used on each. I feel that the user interface can be improved upon, but this additional information would be extremely useful to include in the Elements Manager.

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by panaiotis** [comment]
I hope the Elementor folks are working on his invaluable feature. I have inherited management of several Elementor-based sites that are bloated with different widget packages, some of which use only one instance of one widget from a package. With hundreds of pages and posts to sift through, reducing the bloat becomes nearly impossible.

**Comment by BIGDADDYZJ** [comment]
@panaiotis @micmiller-apptio did you guys ever find a solution to this? I've basically devved this exact feature and can roll this into a plugin if you guys will use it. I know its kinda counterintuitive to be installing more plugins to help with identifying other bloat plugins haha but it is lightweight.

**Comment by panaiotis** [comment]
I never found a solution and it doesn't appear that the folks at Elementor are working on it. I'm interested in this feature and would find a plugin to be handy.

**Reply by micmiller-apptio** [reply]
As mentioned in the original comment. I use a plugin, [Better Usage Statistics for Elementor](https://wordpress.com/plugins/bin-elements-usage-stats-for-elementor). The plugin still works, although it hasn't been updated in three years. The UI is a little clunky (a giant formatted list), but by using search in the web browser I've been able to identify what I need. The capability seems like it would be very safe and easy to add to the Elements Manager (as witnessed by this simple plugin that hasn't needed updating despite numerous updates to Elementor) and a modern UI would be a definite improvement over the Better Usage Statistics plugin.

**Reply by panaiotis** [reply]
I'll try this and see if it does what I need. In a sense, I only need it when I'm evaluating a website I didn't create. Thanks.

**Comment by TRECpro** [comment]
This really is a needed feature. I have over 1,000 blog posts, 3 of which use a gallery widget I need to replace.

**Comment by jorekai** [comment]
If you are stumbling over the same issue feel free to use the following SQL Statement to find the exact post id and location of a widget. You can extract the correct widget type by looking at the HTML Code of the the element manager page. there you can find the data-id of the widget in the table

```SQL
/* 1. All posts, pages, and Elementor templates
that contain the “uicore-woo-product-price” widget */
SELECT  p.ID                                       AS post_id,
        p.post_title                               AS titel,
        p.post_type                                AS typ,
        p.post_status                              AS status,
        pm.meta_value                              AS elementor_json
FROM    wp_posts      AS p
JOIN    wp_postmeta   AS pm  ON pm.post_id = p.ID
WHERE   pm.meta_key   = '_elementor_data'
  AND   pm.meta_value LIKE '%"widgetType":"uicore-woo-product-price"%'
ORDER BY p.post_type, p.post_title;

```
